File: app/src/main/java/com/victoria/bleled/data/model/ModelUser.java
package com.victoria.bleled.data.model;

import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.Ignore;
import androidx.room.PrimaryKey;

@Entity(tableName = "tb_user")
public class ModelUser extends BaseModel {
    @PrimaryKey
    public int uid;

    @ColumnInfo(name = "access_token")
    public String access_token;

    @ColumnInfo(name = "id")
    public String id;

    @ColumnInfo(name = "first_name")
    public String first_name;

    @ColumnInfo(name = "last_name")
    public String last_name;

    @Ignore
    private String reg_time;

    @Ignore
    private String profile_url;

    @Ignore
    private String profile_url_check;

    @Ignore
    private int status;

    public static boolean isNormalUser(String user_id) {
        try {
            if (Integer.parseInt(user_id) > 10) { // 10이하가 특수유저임.
                return true;
            }
        } catch (Exception e) {
            return false;
        }

        return false;
    }

    public enum EUserSex {
        none(""),
        man("m"),
        woman("f");

        private String value;

        EUserSex(String value) {
            this.value = value;
        }

        public String value() {
            return value;
        }

        public static EUserSex toEnum(String value) {
            if (value != null) {
                if (value.equals("m") == true) {
                    return EUserSex.man;
                } else if (value.equals("f") == true) {
                    return EUserSex.woman;
                }
            }

            return EUserSex.none;
        }
    }

    public enum EUserLoginType {
        normal("normal"),
        kakao("kakao"),
        google("google"),
        facebook("facebook");

        private String value;

        EUserLoginType(String value) {
            this.value = value;
        }

        public String value() {
            return value;
        }

        public static EUserLoginType toEnum(String value) {
            if (value != null) {
                if (value.equals("kakao") == true) {
                    return EUserLoginType.kakao;
                } else if (value.equals("google") == true) {
                    return EUserLoginType.google;
                } else if (value.equals("facebook") == true) {
                    return EUserLoginType.facebook;
                }
            }

            return EUserLoginType.normal;
        }
    }
}

File: app/src/main/java/com/victoria/bleled/data/model/ModelUser.kt
package com.victoria.bleled.data.model

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.Ignore
import androidx.room.PrimaryKey

@Entity(tableName = "tb_user")
data class ModelUser(@PrimaryKey(autoGenerate = true) var uid: Int) : BaseModel() {
    @JvmField
    @ColumnInfo(name = "access_token")
    var access_token: String? = null

    @JvmField
    @ColumnInfo(name = "id")
    var id: String? = null

    @JvmField
    @ColumnInfo(name = "first_name")
    var first_name: String? = null

    @JvmField
    @ColumnInfo(name = "last_name")
    var last_name: String? = null

    @Ignore
    private val reg_time: String? = null

    @Ignore
    private val profile_url: String? = null

    @Ignore
    private val profile_url_check: String? = null

    @Ignore
    private val status = 0

    enum class EUserSex(private val value: String) {
        none(""), man("m"), woman("f");

        fun value(): String {
            return value
        }

        companion object {
            fun toEnum(value: String?): EUserSex {
                if (value != null) {
                    if (value == "m" == true) {
                        return man
                    } else if (value == "f" == true) {
                        return woman
                    }
                }
                return none
            }
        }
    }

    enum class EUserLoginType(private val value: String) {
        normal("normal"), kakao("kakao"), google("google"), facebook("facebook");

        fun value(): String {
            return value
        }

        companion object {
            fun toEnum(value: String?): EUserLoginType {
                if (value != null) {
                    if (value == "kakao" == true) {
                        return kakao
                    } else if (value == "google" == true) {
                        return google
                    } else if (value == "facebook" == true) {
                        return facebook
                    }
                }
                return normal
            }
        }
    }

    companion object {
        fun isNormalUser(user_id: String): Boolean {
            try {
                if (user_id.toInt() > 10) { // 10이하가 특수유저임.
                    return true
                }
            } catch (e: Exception) {
                return false
            }
            return false
        }
    }
}